pol-core/bscript/compiler/astbuilder/ExpressionBuilder.cpp

```diff
@@ -38,7 +38,7 @@ ExpressionBuilder::ExpressionBuilder( const SourceFileIdentifier& source_file_id
 {
 }
 
-[[noreturn]] BTokenId ExpressionBuilder::unhandled_operator( const SourceLocation& source_location )
```

Comment:
I'm conflicted on this one. Without `[[noreturn]]` and with `BTokenId`, this method's signature implies that `unhandled_operator` could return a default value. Wouldn't a `[[noreturn]] void` be clearer here?

Reply:
That's the first thing I tried.  This is used in expressions where the operator doesn't match any of the expected values.  If it doesn't have a return type of BTokenId then it doesn't match the type of the expression.

So the options are:
- leave the warning (`[[noreturn BTokenId]`)
- eliminate the warning (but this means it is a function that doesn't ever return, but is not marked `[[noreturn]]` and has a non-void return type)
- redo the expressions to use if-then-else rather than the ternary chain

Reply:
I redid the ternary expressions and went with `[[noreturn]] void`.  Thanks!